### PR TITLE
python3Packages.keras: 2.4.3 -> 2.7.0

### PR DESCRIPTION
**Pull request by veprbl**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by tpwrules** [comment]
It appears pypi no longer has the source for Keras versions > 2.4.3, so this build doesn't actually work?

**Comment by tpwrules** [comment]
It looks like you forgot to add `format = "wheel"`? That hash is correct for the wheel.

**Comment by veprbl** [comment]
> It appears pypi no longer has the source for Keras versions > 2.4.3, so this build doesn't actually work?

It works but only via tarballs.nixos.org
```
# cat test.nix                                                               
with import <nixpkgs> {};

let
  pname = "Keras";
  version = "2.4.3";
in
  python3.pkgs.fetchPypi {
    inherit pname version;
    sha256 = "fedd729b52572fb108a98e3d97e1bac10a81d3917d2103cc20ab2a5f03beb973";
  }
# nix-build test.nix --substituters ""                                             
this derivation will be built:
  /nix/store/k3lvzbprfd3arxy0vl8rljnzw7hb6w1z-Keras-2.4.3.tar.gz.drv
Resolved derivation: '/nix/store/k3lvzbprfd3arxy0vl8rljnzw7hb6w1z-Keras-2.4.3.tar.gz.drv' -> '/nix/store/0b6b3dswcg4zphw7x85apqbk1yrz16a1-Keras-2.4.3.tar.gz.drv'...
building '/nix/store/0b6b3dswcg4zphw7x85apqbk1yrz16a1-Keras-2.4.3.tar.gz.drv'...

trying http://tarballs.nixos.org/sha256/fedd729b52572fb108a98e3d97e1bac10a81d3917d2103cc20ab2a5f03beb973
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100  284k  100  284k    0     0  2732k      0 --:--:-- --:--:-- --:--:-- 2732k
/nix/store/6kb2ifkq9fczkscmwfk6d0526rafdpam-Keras-2.4.3.tar.gz
```

**Comment by yl3dy** [comment]
Result of `nixpkgs-review pr 147418` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>14 packages marked as broken and skipped:</summary>
  <ul>
    <li>python38Packages.arviz</li>
    <li>python38Packages.dm-sonnet</li>
    <li>python38Packages.graph_nets</li>
    <li>python38Packages.optuna</li>
    <li>python38Packages.pymc3</li>
    <li>python38Packages.rl-coach</li>
    <li>python38Packages.tensorflow-probability</li>
    <li>python39Packages.arviz</li>
    <li>python39Packages.dm-sonnet</li>
    <li>python39Packages.graph_nets</li>
    <li>python39Packages.optuna</li>
    <li>python39Packages.pymc3</li>
    <li>python39Packages.rl-coach</li>
    <li>python39Packages.tensorflow-probability</li>
  </ul>
</details>
<details>
  <summary>20 packages failed to build:</summary>
  <ul>
    <li>gpt2tc</li>
    <li>python38Packages.baselines</li>
    <li>python38Packages.edward</li>
    <li>python38Packages.einops</li>
    <li>python38Packages.mask-rcnn</li>
    <li>python38Packages.scikit-tda</li>
    <li>python38Packages.tensorflow (python38Packages.tensorflow-build ,python38Packages.tensorflowWithoutCuda)</li>
    <li>python38Packages.tensorflowWithCuda</li>
    <li>python38Packages.tflearn</li>
    <li>python38Packages.umap-learn</li>
    <li>python39Packages.baselines</li>
    <li>python39Packages.edward</li>
    <li>python39Packages.einops</li>
    <li>python39Packages.mask-rcnn</li>
    <li>python39Packages.scikit-tda</li>
    <li>python39Packages.tensorflow (python39Packages.tensorflow-build ,python39Packages.tensorflowWithoutCuda)</li>
    <li>python39Packages.tensorflowWithCuda</li>
    <li>python39Packages.tflearn</li>
    <li>python39Packages.umap-learn</li>
    <li>tts</li>
  </ul>
</details>
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python38Packages.keras</li>
    <li>python39Packages.keras</li>
  </ul>
</details>

**Comment by veprbl** [comment]
```
14 packages marked as broken and skipped:
python38Packages.arviz python38Packages.dm-sonnet python38Packages.graph_nets python38Packages.optuna python38Packages.pymc3 python38Packages.rl-coach python38Packages.tensorflow-probability python39Packages.arviz python39Packages.dm-sonnet python39Packages.graph_nets python39Packages.optuna python39Packages.pymc3 python39Packages.rl-coach python39Packages.tensorflow-probability
7 packages failed to build:
python38Packages.einops python38Packages.scikit-tda python38Packages.umap-learn python39Packages.einops python39Packages.scikit-tda python39Packages.umap-learn tts
15 packages built:
gpt2tc python38Packages.baselines python38Packages.edward python38Packages.keras python38Packages.mask-rcnn python38Packages.tensorflow python38Packages.tensorflowWithCuda python38Packages.tflearn python39Packages.baselines python39Packages.edward python39Packages.keras python39Packages.mask-rcnn python39Packages.tensorflow python39Packages.tensorflowWithCuda python39Packages.tflearn
```

**Comment by veprbl** [comment]
The `einops` is broken on Hydra. The `umap-learn` is probably broken by this PR, but it's also broken in the same way in the upstream CI build: https://dev.azure.com/lelandmcinnes/UMAP%20project%20builds/_build/results?buildId=507&view=logs&j=a7a13765-38c1-5bc9-e079-697f9322f239&t=bbe0dc9f-c6da-5876-be31-7c63bba3316e
